### PR TITLE
Android gradle plugin 3 alpha

### LINKED ISSUE #40: Gradle 3.0 alpha 5, Android Studio 3.0 preview
Thank you for all your work so far. It helped us a lot with our products. 
I'm investigating the new Android Studio 3.0 preview with Gradle 3.0.0-alpha5 but I have an error building our app.
I'm using version 2.3.2 of this plugin.

> Error:Could not get unknown property 'manifestOutputFile' for task ':app:processDebugManifest' of type com.android.build.gradle.tasks.MergeManifests.

When I disable this plugin, the gradle build does work again.
Thanks in advance!

### PR DESCRIPTION
**Pull request by fractalwrench**
Fixes breaking changes in the Variant API which were introduced in 3.0.0-beta3 of the Gradle plugin. Changes include:

- Updated gradle dependencies/gradle wrapper to latest versions
- Increase logging
- Refactor 3 tasks into methods (setup autoConfig of proguard, adding build UUID to manifest, uploading mapping file to server)
- Update build UUID/mapping file upload to execute in `output.processManifest.doLast`, as the manifest is no longer [guaranteed to be created during the configuration phase](https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#variant_api).

The Android plugin for gradle is currently in beta, so we'd need to mark any release as beta, as it contains breaking changes for anyone still on 2.X.

Fixes #40

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
     id 'com.bmuschko.nexus' version '2.3.1'
-    id 'com.jfrog.bintray' version '1.7'
+    id 'com.jfrog.bintray' version '1.7.3'
 }
 apply plugin: 'maven-publish'
 
 // Repositories for dependencies
 repositories {
     jcenter()
+    mavenCentral()
```
Comment (artem-zinnatullin) [inline build.gradle]: Why do you need `mavenCentral()`? 

`jcenter()` proxies `mavenCentral()` :)

File: build.gradle [hunk 2]
```diff
@@ -22,9 +24,9 @@ compileGroovy {
 dependencies {
     compile gradleApi()
     compile localGroovy()
-    compile 'com.android.tools.build:gradle:2.2.0'
-    compile 'org.apache.httpcomponents:httpclient:4.2.1'
-    compile 'org.apache.httpcomponents:httpmime:4.2.1'
+    compile 'com.android.tools.build:gradle:3.0.0-beta3'
```
Comment (artem-zinnatullin) [inline build.gradle]: yayayayay!

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Wed Apr 10 15:27:10 PDT 2013
+#Thu Aug 03 16:43:15 BST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-rc-1-all.zip
```
Comment (artem-zinnatullin) [inline gradle/wrapper/gradle-wrapper.properties]: 4.1 is already stable release https://docs.gradle.org/4.1/release-notes.html !

**Comment by fractalwrench** [comment]
thanks for the review comments @artem-zinnatullin!

**Comment by artem-zinnatullin** [comment]
Pff, thank you for supporting AGP beta!

**Pull request merged**